hack/test-go.sh

```diff
@@ -54,6 +55,11 @@ go_test_flags=(
 packages=()
 mapfile -t packages < <(go list ./... | grep -v 'sigs.k8s.io/k8s-container-image-promoter/cmd\|test-e2e')
 
-GO111MODULE=on go test "${go_test_flags[@]}" "${packages[@]}"
+for p in "${packages[@]}"; do
+    if ! GO111MODULE=on go test "${go_test_flags[@]}" "${p}"; then
+        # Exit early.
+        exit 1
+    fi
+done
 
 go tool cover -html "${ARTIFACTS}/coverage.out" -o "${ARTIFACTS}/coverage.html"
```

Comment:
Then maybe a counter of sorts to increment on the exit statuses of each function run, _after_ `go tool cover`.
If the number of `return 1`s is greater than zero, then `exit 1`.